src/tensors/factorizations.jl

```diff
@@ -222,10 +222,7 @@ matrices. See the corresponding documentation for more information.
 
 See also `eigen` and `eigh`.
 """
-function eig(t::AbstractTensorMap, p::Index2Tuple; kwargs...)
-    tcopy = permutedcopy_oftype(t, factorisation_scalartype(eig, t), p)
-    return eig!(tcopy; kwargs...)
-end
+eig(t::AbstractTensorMap, p::Index2Tuple; kwargs...) = MAK.eig_full(t; kwargs...)
```

Comment:
This seems to loose the `p` argument. In first instance, I would perhaps only route the `factorization!(t)` methods through `MAK`, and then gradually extend from there.